src/toil/provisioners/clusterScaler.py

```diff
@@ -235,6 +235,30 @@ def __init__(self, provisioner, jobBatcher, config):
         else:
             self.scaler = None
 
+    def check(self):
+        """
+        Attempt to join any existing scaler threads with a short timeout. This insures an exception
+        raised in the threads is propagated in a timely fashion. If there are exceptions in both threads,
```

Comment:
I kept the original wording here since my new approach doesn't actually raise the errors, just logs them